packet/flatcar-linux/kubernetes/calico/host-protection.yaml.tmpl

```diff
@@ -1,126 +1,109 @@
-# Customize inbound failsafe rules
-# https://docs.projectcalico.org/v3.5/getting-started/bare-metal/policy/failsafe
+# General
+# =======
+---
+# Disable default failsafe inbound rules
+apiVersion: crd.projectcalico.org/v1
+kind: FelixConfiguration
+metadata:
+  name: default
+spec:
+  failsafeInboundHostPorts: []
+
+# Host endpoints
+# ==============
+---
+${controller_host_endpoints}
+---
+${worker_host_endpoints}
+
+# Firewall policy
+# ===============
+---
+# SSH access
 apiVersion: crd.projectcalico.org/v1
 kind: GlobalNetworkPolicy
 metadata:
-  name: failsafe-ingress
+  name: ssh
 spec:
+  selector: has(host-endpoint)
   preDNAT: true
   applyOnForward: true
-  selector: has(host-endpoint)
   order: 0
   ingress:
-  # SSH access
   - action: Allow
     protocol: TCP
     source:
       nets: ${management_cidrs}
     destination:
       ports: [22]
-  # DHCP access
-  - action: Allow
-    protocol: UDP
-    source:
-      nets: ${node_public_ips}
-    destination:
-      ports: [68]
-  # BGP and etcd access
-  - action: Allow
-    protocol: TCP
-    source:
-      nets: ${node_public_ips}
-    destination:
-      ports:
-      # BGP access (Calico networking)
-      - 179
-      # etcd access
-      - 2379
-      - 2380
-      # etcd self-hosted service access
-      - 6666
-      - 6667
 ---
-# Allow traffic from IP addresses that are known to be cluster-internal including pods and services
+# kubectl access
 apiVersion: crd.projectcalico.org/v1
 kind: GlobalNetworkPolicy
 metadata:
-  name: allow-cluster-internal-ingress
+  name: kubectl
 spec:
-  order: 10
+  selector: nodetype == 'controller'
   preDNAT: true
   applyOnForward: true
+  order: 0
   ingress:
-    - action: Allow
-      source:
-        nets: ${cluster_internal_cidrs}
-  selector: has(host-endpoint)
+  - action: Allow
+    protocol: TCP
+    destination:
+      ports: [6443]
 ---
-# Deny traffic from any other sources
+# Traffic between cluster nodes
 apiVersion: crd.projectcalico.org/v1
 kind: GlobalNetworkPolicy
 metadata:
-  name: drop-other-ingress
+  name: inter-node-traffic
 spec:
-  order: 20
+  selector: has(host-endpoint)
   preDNAT: true
   applyOnForward: true
-  ingress:
-    - action: Deny
-  selector: has(host-endpoint)
----
-# Allow egress traffic through nodes' external interface
-apiVersion: crd.projectcalico.org/v1
-kind: GlobalNetworkPolicy
-metadata:
-  name: allow-outbound-external
-spec:
   order: 10
-  egress:
+  ingress:
     - action: Allow
-  selector: has(host-endpoint)
----
-# controller hosts endpoint
-${controller_host_endpoints}
----
-# worker hosts endpoint
-${worker_host_endpoints}
+      source:
+        nets: ${cluster_internal_cidrs}
```

Comment:
Should we allow all kind of traffic between nodes?

Reply:
We can make this more strict, but we would need to research what traffic to allow. I suggest opening a separate issue for this as my main motivation in this PR was to remove unnecessary rules and make things easier to maintain, not necessarily make security stricter.